autogen/oai/client.py

```diff
@@ -161,6 +169,12 @@ def create(self, params: Dict[str, Any]) -> ChatCompletion:
 
         completions: Completions = self._oai_client.chat.completions if "messages" in params else self._oai_client.completions  # type: ignore [attr-defined]
         # If streaming is enabled and has messages, then iterate over the chunks of the response.
+
+        _is_vision = is_vision_model(params.get("model", None))
+        _is_vision |= params.get("vision_model", False)
+        if _is_vision and "messages" in params:
+            params["messages"] = format_message_contents_with_images(params["messages"])
```

Comment:
This should be a noop for non image related messages. We should just always do this transformation to avoid needing to check and specify

Reply:
One issue if we always perform this transformation is what if the user sends images to a non-vision model. We can add some check and warnings here if this happens.

Now, if user sends images to a non-vision model, it will not raise any errors here, but the LLM might say "I am not able to view the images".

Reply:
Do you mean before or after the change the LLM responds with "I am not able to view the images"?

Reply:
The model will say "I am not able to view the images" now.

If we transform messages regardless for all models, OpenAI will throw an exception: `BadRequestError`. 

So, I would suggest not transforming the message format unless it will communicate for a multimodal client.